pkg/controller/tenants.go

```diff
@@ -45,6 +60,162 @@ func (c *Controller) getTenantConfiguration(ctx context.Context, tenant *miniov2
+	tenantConfiguration["MINIO_UPDATE_MINISIGN_PUBKEY"] = "RWTx5Zr1tiHQLwG9keckT0c45M3AGeHD6IvimQHpyRywVWGbP1aVSGav"
+	tenantConfiguration["MINIO_OPERATOR_VERSION"] = c.operatorVersion
+	tenantConfiguration["MINIO_PROMETHEUS_JOB_ID"] = tenant.PrometheusConfigJobName()
+
+	var domains []string
+	// Enable Bucket DNS only if asked for by default turned off
+	if tenant.BucketDNS() {
+		domains = append(domains, tenant.MinIOBucketBaseDomain())
+		sidecarBucketURL := fmt.Sprintf("http://127.0.0.1:%s%s/%s/%s",
+			common.WebhookDefaultPort,
+			common.WebhookAPIBucketService,
+			tenant.Namespace,
+			tenant.Name)
+		tenantConfiguration[common.BucketDNSEnv] = sidecarBucketURL
+	}
+	// Check if any domains are configured
+	if tenant.HasMinIODomains() {
+		domains = append(domains, tenant.GetDomainHosts()...)
+	}
+	// tell MinIO about all the domains meant to hit it if they are not passed manually via .spec.env
+	if len(domains) > 0 {
+		tenantConfiguration[miniov2.MinIODomain] = strings.Join(domains, ",")
+	}
+	// If no specific server URL is specified we will specify the internal k8s url, but if a list of domains was
+	// provided we will use the first domain.
+	serverURL := tenant.MinIOServerEndpoint()
+	if tenant.HasMinIODomains() {
+		// Infer schema from tenant TLS, if not explicit
+		if !strings.HasPrefix(tenant.Spec.Features.Domains.Minio[0], "http") {
+			useSchema := "http"
+			if tenant.TLS() {
+				useSchema = "https"
+			}
+			serverURL = fmt.Sprintf("%s://%s", useSchema, tenant.Spec.Features.Domains.Minio[0])
+		} else {
+			serverURL = tenant.Spec.Features.Domains.Minio[0]
+		}
+	}
+	tenantConfiguration[miniov2.MinIOServerURL] = serverURL
+
+	// Set the redirect url for console
+	if tenant.HasConsoleDomains() {
+		consoleDomain := tenant.Spec.Features.Domains.Console
+		// Infer schema from tenant TLS, if not explicit
+		if !strings.HasPrefix(consoleDomain, "http") {
+			useSchema := "http"
+			if tenant.TLS() {
+				useSchema = "https"
+			}
+			consoleDomain = fmt.Sprintf("%s://%s", useSchema, consoleDomain)
+		}
+		tenantConfiguration[miniov2.MinIOBrowserRedirectURL] = consoleDomain
+	}
+
+	if tenant.HasKESEnabled() {
+		tenantConfiguration["MINIO_KMS_KES_ENDPOINT"] = tenant.KESServiceEndpoint()
+		tenantConfiguration["MINIO_KMS_KES_CERT_FILE"] = miniov2.MinIOCertPath + "/client.crt"
+		tenantConfiguration["MINIO_KMS_KES_KEY_FILE"] = miniov2.MinIOCertPath + "/client.key"
+		tenantConfiguration["MINIO_KMS_KES_CA_PATH"] = miniov2.MinIOCertPath + "/CAs/kes.crt"
+		tenantConfiguration["MINIO_KMS_KES_KEY_NAME"] = tenant.Spec.KES.KeyName
+	}
+
+	// Set the env variables in the tenant.spec.env field
+	// User defined environment variables will take precedence over default environment variables
+	envVars := tenant.GetEnvVars()
+	for _, ev := range envVars {
+		tenantConfiguration[ev.Name] = ev.Value
+	}
+
+	configurationSecret, err := c.kubeClientSet.CoreV1().Secrets(tenant.Namespace).Get(ctx, tenant.Spec.Configuration.Name, metav1.GetOptions{})
```

Comment:
I don't think we should overwrite the declarative secret the users put in, this should be generated on the fly and placed in the tmp we do in the sidecar from the sidecar alone

Reply:
okay, Let me sort it out, will be ready in next commit